### PR TITLE
Model for Audit Log

### PR DESCRIPTION
**Pull request by damnhipster**
#### Background

Model for audit log.

- mapped to the table schema
- the delete function has been disabled
- Pubsweet's base schema requires an `updated` column, and there doesn't seem to be a way to avoid having one. Added a migration to add the required column

#### Any relevant tickets

#1110 

#### How has this been tested?

Please indicate the need for this change to be tested. If not please justify, if so then at what levels. **Reviewers** ensure that these test requirements are also reviewed.

- [x] Unit / Integration tests
- [ ] Browser tests
- [ ] End2end tests

### CODE REVIEW

File: server/xpub-model/entities/auditLog/index.js [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        userId: { type: 'uuid' },
+        action: { type: 'string' },
+        objectId: { type: 'uuid' },
+        objectType: { type: ['string', 'null'] }
+      },
+    }
+  }
+
+  async delete() {
+    throw new Error('Unsupported operation')
```
Comment (diversemix) [inline server/xpub-model/entities/auditLog/index.js]: Nice! ... do we want one for `update` too?

File: server/xpub-model/migrations/1545390290-alter-audit-log.sql [hunk 2]
```diff
@@ -0,0 +1,2 @@
+ALTER TABLE audit_log
+  ADD COLUMN updated TIMESTAMP WITH TIME ZONE
```
[the remark this answers is not in the document]
Comment (damnhipster) [reply server/xpub-model/migrations/1545390290-alter-audit-log.sql]: We can override the `save` function to not allow updates on rows.

File: server/xpub-model/entities/auditLog/index.js (hunk 1 above)
Comment (damnhipster) [reply server/xpub-model/entities/auditLog/index.js]: `delete` is provided by Pubsweet, which is why I took the opportunity to disable it. `save` is the only other method that is provided which has write access, other than the base methods provided by Objection.

File: server/xpub-model/migrations/1545390290-alter-audit-log.sql (hunk 2 above)
[the remark this answers is not in the document]
Comment (damnhipster) [reply server/xpub-model/migrations/1545390290-alter-audit-log.sql]: Added requirement to #1110

**Pull request merged**